src/core/shape_inference/include/fft_base_shape_inference.hpp

```diff
@@ -53,14 +53,23 @@ void shape_infer(const ov::op::util::FFTBase* op,
         // [n_0, ..., n_{r - 1}, 2] is interpreted as a complex tensor with the shape
         // [n_0, ..., n_{r - 1}].
         if (axes_shape.rank().is_static() && axes_are_known) {
+            ov::AxisSet axes_set;
             for (int64_t& axis : axes) {
+                NODE_VALIDATION_CHECK(op,
+                                      axis < static_cast<int64_t>(input_rank),
```

Comment:
I'd rather perform the whole range validation in this place. This way the error message can contain information about the bounds in which the axis value must fit. The two separate messages below indicate that the axis value must implicitly be positive which contradicts with the docs.
```suggestion
-static_cast<int64_t>(input_rank) <= axis && axis < static_cast<int64_t>(input_rank),
```

Reply:
done